README.md
# custom_radio_button

[![pub package](https://img.shields.io/pub/v/custom_radio_button.svg)](https://pub.dartlang.org/packages/custom_radio_button)

A Flutter package for iOS and Android for Custom Radio Button.

## Demo
<img src="http://sheikhsoft.com/screensort/custom_radio_button_1.png" width="340" height="640" title="Screen Shoot">
<img src="http://sheikhsoft.com/screensort/custom_radio_button_2.png" width="340" height="640" title="Screen Shoot">

## Installation

First, add `custom_radio_button` as a [dependency in your pubspec.yaml file](https://flutter.io/platform-plugins/).

### iOS

No configuration required - the plugin should work out of the box.

### Android

No configuration required - the plugin should work out of the box.

### Code for the Square Radio Group widget

``` dart
import 'package:custom_radio_button/custom_radio_button.dart';
import 'package:custom_radio_button/radio_model.dart';

class MyHomePage extends StatefulWidget {
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  String _selectedValue;
  
 
  @override
  Widget build(BuildContext context) {
  List<RadioModel> priorityList = new List<RadioModel>();
        priorityList.add(new RadioModel(false, null, 'Heigh', Colors.redAccent));
        priorityList
            .add(new RadioModel(false, null, 'Medium', Colors.deepPurple));
        priorityList
            .add(new RadioModel(false, null, 'Slight', Colors.blueAccent));
        priorityList
            .add(new RadioModel(false, null, 'Normal', Colors.purpleAccent));
    return CustomRadioGroupWidget(
                           onChanged: getvalue,
                           isSquareRadioGroup: true,
                           radioList: priorityList,
                         );
  }
  void getvalue(Map value) {
      _selectedValue = value;
    }
}
```

### Code for the Round Radio Group widget

``` dart
import 'package:custom_radio_button/custom_radio_button.dart';
import 'package:custom_radio_button/radio_model.dart';

class MyHomePage extends StatefulWidget {
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  String _selectedValue;
  
 
  @override
  Widget build(BuildContext context) {
  List<RadioModel> incidentTypeList = new List<RadioModel>();
        incidentTypeList.add(new RadioModel(
            false,
            Icon(
              Icons.whatshot,
              color: Colors.blue,
            ),
            'Fire',
            Colors.red));
        incidentTypeList.add(new RadioModel(
            false,
            Icon(
              Icons.face,
              color: Colors.red,
            ),
            'Near Miss',
            Colors.deepPurple));
        incidentTypeList.add(new RadioModel(
            false,
            Icon(
              Icons.directions_car,
              color: Colors.deepOrange,
            ),
            'Accident',
            Colors.blueAccent));
        incidentTypeList.add(new RadioModel(
            false,
            Icon(
              Icons.directions_run,
              color: Colors.greenAccent,
            ),
            'Theft',
            Colors.purpleAccent));
        incidentTypeList.add(new RadioModel(
            false,
            Icon(
              Icons.business,
              color: Colors.yellow,
            ),
            'Property damage',
            Colors.greenAccent));
    return CustomRadioGroupWidget(
                           onChanged: getvalue,
                           isSquareRadioGroup: false,
                           radioList: incidentTypeList,
                         );
  }
  void getvalue(Map value) {
      _selectedValue = value;
    }
}
```

### Code for the Full page Radio Group Widget

``` dart
import 'package:custom_radio_button/full_page_radio_group.dart';

class RadioGroupPage extends StatelessWidget {
  final Map mapData;
  const RadioGroupPage({Key key, this.mapData}) : super(key: key);

  @override
  Widget build(BuildContext context) {
   

    return new FullPageRadioGroup(
        
        isSquareRadioGroup: false,
        title: "What happended?",
        route: "/fourth",
        mapData: mapData,
        mapKey: 'incidentType');
  }
}
```

CHANGELOG.md
## [0.0.1]
* Initial release



lib/custom_radio_button.dart
library custom_radio_button;

import 'package:flutter/material.dart';

import 'radio_model.dart';

class CustomRadioGroupWidget extends StatefulWidget {
  final ValueChanged<String> onChanged;
  final bool isSquareRadioGroup;
  final List<RadioModel> radioList;

  const CustomRadioGroupWidget(
      {Key key, this.isSquareRadioGroup, this.radioList, this.onChanged})
      : super(key: key);

  @override
  _CustomRadioGroupWidgetState createState() => _CustomRadioGroupWidgetState();
}

class _CustomRadioGroupWidgetState extends State<CustomRadioGroupWidget> {
  String _selectedValue;
  @override
  void initState() {
    // TODO: implement initState
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return Container(
      alignment: Alignment.center,
      margin: EdgeInsets.only(top: 50.0),
      child: widget.isSquareRadioGroup
          ? _buildSquareRadioList()
          : _buildRoundRadioGroup(),
    );
  }

  Widget _buildSquareRadioList() {
    return Container(
      decoration: new BoxDecoration(
          boxShadow: [
            BoxShadow(
              color: Colors.grey,
              blurRadius: 5.0, // has the effect of softening the shadow
              spreadRadius: 3.0, // has the effect of extending the shadow
              offset: Offset(
                5.0, // horizontal, move right 10
                5.0, // vertical, move down 10
              ),
            )
          ],
          borderRadius: const BorderRadius.all(const Radius.circular(50.0)),
          color: Colors.white),
      width: 250.0,
      height: 360.0,
      child: ClipRRect(
        borderRadius: const BorderRadius.all(const Radius.circular(50.0)),
        child: new ListView.builder(
          itemCount: widget.radioList.length,
          itemBuilder: (BuildContext context, int index) {
            return new InkWell(
              onTap: () {
                setState(() {
                  widget.radioList
                      .forEach((element) => element.isSelected = false);
                  widget.radioList[index].isSelected = true;
                  _selectedValue = widget.radioList[index].text;
                  _publishSelection(_selectedValue);
                });
              },
              child: new SquareRadioItem(widget.radioList[index]),
            );
          },
        ),
      ),
    );
  }

  Widget _buildRoundRadioGroup() {
    return new ListView.builder(
      itemCount: widget.radioList.length,
      itemBuilder: (BuildContext context, int index) {
        return new InkWell(
          onTap: () {
            setState(() {
              widget.radioList.forEach((element) => element.isSelected = false);
              widget.radioList[index].isSelected = true;
              _selectedValue = widget.radioList[index].text;
              _publishSelection(_selectedValue);
            });
          },
          child: new RoundRadioItem(widget.radioList[index]),
        );
      },
    );
  }

  void _publishSelection(selectedValue) {
    if (widget.onChanged != null) {
      widget.onChanged(selectedValue);
    }
  }
}

class SquareRadioItem extends StatelessWidget {
  final RadioModel _item;
  SquareRadioItem(this._item);
  @override
  Widget build(BuildContext context) {
    return Container(
      height: 80.0,
      width: double.infinity,
      child: new Center(
          child: Row(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              _item.radioIcon != null
                  ? IconButton(
                icon: _item.radioIcon,
                onPressed: null,
              )
                  : Container(),
              new Text(_item.text,
                  style: new TextStyle(
                      color: _item.isSelected ? Colors.white : Colors.black,
                      //fontWeight: FontWeight.bold,
                      fontSize: 18.0,
                      fontWeight: FontWeight.bold)),
            ],
          )),
      decoration: new BoxDecoration(
        color: _item.isSelected ? _item.selectedColor : Colors.white,
        border: new Border.all(
            width: 1.0,
            color: _item.isSelected ? _item.selectedColor : Colors.white),
      ),
    );
  }
}

class RoundRadioItem extends StatelessWidget {
  final RadioModel _item;

  RoundRadioItem(this._item);

  @override
  Widget build(BuildContext context) {
    return Container(
      margin:
      new EdgeInsets.only(left: 25.0, right: 25.0, top: 8.0, bottom: 8.0),
      height: 80.0,
      width: double.infinity,
      child: new Center(
          child: Row(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              _item.radioIcon != null
                  ? IconButton(
                icon: _item.radioIcon,
                onPressed: null,
              )
                  : Container(),
              new Text(_item.text,
                  style: new TextStyle(
                      color: _item.isSelected ? Colors.white : Colors.black,
                      //fontWeight: FontWeight.bold,
                      fontSize: 18.0,
                      fontWeight: FontWeight.bold)),
            ],
          )),
      decoration: new BoxDecoration(
        boxShadow: [
          BoxShadow(
            color: Colors.grey,
            blurRadius: 10.0, // has the effect of softening the shadow
            spreadRadius: 3.0, // has the effect of extending the shadow
            offset: Offset(
              5.0, // horizontal, move right 10
              5.0, // vertical, move down 10
            ),
          )
        ],
        color: _item.isSelected ? _item.selectedColor : Colors.white,
        border: new Border.all(
            width: 1.0,
            color: _item.isSelected ? _item.selectedColor : Colors.white),
        borderRadius:
        const BorderRadius.all(const Radius.elliptical(100.0, 100.0)),
      ),
    );
  }
}


lib/full_page_radio_group.dart
library custom_radio_button;

import 'package:custom_radio_button/utils.dart';
import 'package:flutter/material.dart';

import 'custom_radio_button.dart';
import 'radio_model.dart';


class FullPageRadioGroup extends StatelessWidget {
  Map mapData;
  final Widget container;
  final String title;
  final String route;
  List<RadioModel> radioList;
  final bool isSquareRadioGroup;
  final String mapKey;

  FullPageRadioGroup(
      {Key key,
      this.mapData,
      this.container,
      this.title,
      this.route,
      this.radioList,
      this.isSquareRadioGroup,
      this.mapKey})
      : super(key: key);

  String _selectedValue;

  @override
  Widget build(BuildContext context) {
    if (radioList == null) {
      List<RadioModel> priorityList = new List<RadioModel>();
      priorityList.add(new RadioModel(false, null, 'Heigh', Colors.redAccent));
      priorityList
          .add(new RadioModel(false, null, 'Medium', Colors.deepPurple));
      priorityList
          .add(new RadioModel(false, null, 'Slight', Colors.blueAccent));
      priorityList
          .add(new RadioModel(false, null, 'Normal', Colors.purpleAccent));

      List<RadioModel> incidentTypeList = new List<RadioModel>();
      incidentTypeList.add(new RadioModel(
          false,
          Icon(
            Icons.whatshot,
            color: Colors.blue,
          ),
          'Fire',
          Colors.red));
      incidentTypeList.add(new RadioModel(
          false,
          Icon(
            Icons.face,
            color: Colors.red,
          ),
          'Near Miss',
          Colors.deepPurple));
      incidentTypeList.add(new RadioModel(
          false,
          Icon(
            Icons.directions_car,
            color: Colors.deepOrange,
          ),
          'Accident',
          Colors.blueAccent));
      incidentTypeList.add(new RadioModel(
          false,
          Icon(
            Icons.directions_run,
            color: Colors.greenAccent,
          ),
          'Theft',
          Colors.purpleAccent));
      incidentTypeList.add(new RadioModel(
          false,
          Icon(
            Icons.business,
            color: Colors.yellow,
          ),
          'Property damage',
          Colors.greenAccent));
      if (mapKey == "priority") {
        radioList = priorityList;
      } else if (mapKey == "incidentType") {
        radioList = incidentTypeList;
      }
    }
    return Scaffold(
      body: new Container(
        child: Stack(
          children: <Widget>[
            container != null ? container : Container(),
            TopTitle(
              topMargin: 80.0,
              leftMargin: 50.0,
              title: title,
            ),
            Container(
              alignment: Alignment.center,
              margin: EdgeInsets.only(top: 50.0),
              child: CustomRadioGroupWidget(
                onChanged: getPriority,
                isSquareRadioGroup: isSquareRadioGroup,
                radioList: radioList,
              ),
            ),
            Container(
              padding: EdgeInsets.all(10.0),
              alignment: Alignment.bottomRight,
              child: FloatingActionButton(
                backgroundColor: Colors.black87,
                child: Icon(Icons.arrow_forward),
                onPressed: () {
                  print(_selectedValue);

                  if (_selectedValue != null) {
                    if (mapData == null) {
                      Map newMap = new Map();
                      newMap[mapKey] = _selectedValue;

                      Navigator.pushNamed(context, route, arguments: newMap);
                    } else {
                      mapData[mapKey] = _selectedValue;

                      Navigator.pushNamed(context, route, arguments: mapData);
                    }
                  } else {
                    Utils().showMyDialog(context, "Please Select A Option");
                  }
                },
              ),
            ),
            MyBackButton(),
          ],
        ),
      ),
    );
  }

  void getPriority(String value) {
    _selectedValue = value;
  }
}
